core/src/main/java/org/infinispan/context/EntryLookup.java

```diff
@@ -28,15 +31,28 @@
     * Note: The key inside the {@linkplain CacheEntry} may be {@code null} if the key does not exist in the cache.
     *
     * @return a map of looked up entries.
-    *
-    * @deprecated Since 9.3, please use {@link #forEachEntry(BiConsumer)} or {@link #lookedUpEntriesCount()} instead.
+    * @deprecated Since 9.3, please use {@link #forEachEntry(BiConsumer)}, {@link #lookedUpEntriesCount()} or
+    * {@link #publisher()} instead.
     */
    @Deprecated
    Map<Object, CacheEntry> getLookedUpEntries();
 
    /**
-    * Execute an action for each value in the context.
+    * Returns a Publisher that when subscribed to provide all values that have a value in the given context.
     *
+    * @param <K> key type provided from user
+    * @param <V> value type provided from user
+    * @return
+    */
+   default <K, V> Publisher<CacheEntry<K, V>> publisher() {
```

Comment:
I was thinking of moving this implementation in `AbstractInvocationContext` and implementing the method in e.g. `SingleNonTxInvocationContext` to return `Flowable.empty()` if `entry == null || entry.isNull()` and `Flowable.just(entry)` otherwise.

Reply:
I can still move this to `AbstractInvocationContext` though and not have the default method.